### ISSUE TITLE
question regarding output channel change

### ISSUE THREAD

**Issue by machinehistories** [adafruit/Adafruit_CircuitPython_MIDI#50]
I am trying to change the midi output channel dynamically and I've tried a few different things but all of them lock up as soon as I attempt to change the midi out channel. I know this probably isn't the best place to ask this so I apologize if I should be asking for support elsewhere. In the docs it sounds like this can be done. For example in the documentation for control change it states.
Parameters
control – The control number, 0-127.

value – The 7bit value of the control, 0-127.

channel – The channel number of the MIDI message where appropriate. This is updated by MIDI.send() method.

However cc only takes 2 arguments so channel can't be sent there. If I try to redefine macropad = MacroPad(rotation=90, midi_in_channel=1, midi_out_channel=dynamic) it doesn;t like it and if I try macropad.midi.send(macropad.midi_out_channel(dynamic+1)) it doesn't like it. Any advice on this would be great. It seems on the op-z you need a channel change and not just a cc to switch tracks and that is why I want to be able to redefine the midi out channel dynamically.

**Comment by todbot** [comment]
The `midi_out_channel` param in the `MacroPad` constructor sets the default MIDI channel that `macropad.midi.send()` will use if no channel is given.  

If at the time of sending you want to use a different channel,  you can provide that as an argument to `send()` like:  ` macropad.midi.send( macropad.NoteOn(note, 124), channel=chan)`

As an example, here's a program that uses the keys to send MIDI notes, with the encoder selecting which channel to send on:

```py
import time
import adafruit_macropad

macropad = adafruit_macropad.MacroPad()
base_note = 36

while True:
    key_event = macropad.keys.events.get()
    if key_event:
        if key_event.pressed:
            note = base_note + key_event.key_number
            chan = (macropad.encoder % 16)  # constrain to 0-15
            print("sening note",note, "on MIDI channel", (chan+1))
            macropad.midi.send( macropad.NoteOn(note, 124), channel=chan)
            time.sleep(0.25)
            macropad.midi.send( macropad.NoteOff(note, 123), channel=chan)
```

Note that, confusingly, there is also a "channel" property in the `NoteOn` and `NoteOff` objects. Those are overwritten by either `channel=` in `send()` or by the default `midi_out_channel`.

**Comment by machinehistories** [comment]
Oh my this is fantastic thanks so much for taking the time to explain this. I have made some modifications to the macropad code to help dynamically select cc channel and then use the keys to send out values in the proper range for various op-z controls. I found some great code that I incorporated to keep timing synced and this channel control was the last feature I really wanted to add. I can't thank you enough for the library it's great fun and it allows me to take control of the op-z in a way I couldn't otherwise and it's so fun to think of ways to make a single button/dial ui do all sorts of different things. Again many thanks to adafruit for all the hard work you all do.
